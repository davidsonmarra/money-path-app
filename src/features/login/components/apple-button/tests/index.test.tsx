import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Button from 'src/features/login/components/apple-button';

interface Props {
  onPress?: () => void;
}

const mockText = 'Entrar com Apple';

const renderComponent = ({onPress}: Props) =>
  render(<Button onPress={onPress} />);

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    renderComponent({});
    const button = screen.getByText(mockText);
    expect(button.children).toEqual([mockText]);
  });

  it('should call onPress', () => {
    const onPress = jest.fn();
    const button = renderComponent({onPress}).root;
    button.props.onClick();
    expect(onPress).toHaveBeenCalled();
  });
});
