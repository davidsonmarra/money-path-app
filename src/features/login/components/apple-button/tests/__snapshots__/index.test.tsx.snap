// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 12,
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
>
  <View>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={28}
      bbWidth={28}
      fill="#FFF"
      focusable={false}
      height={28}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 28,
            "width": 28,
          },
        ]
      }
      testID="icon-apple"
      vbHeight={22.773}
      vbWidth={22.773}
      width={28}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M15.769 0h.162c.13 1.606-.483 2.806-1.228 3.675-.731.863-1.732 1.7-3.351 1.573-.108-1.583.506-2.694 1.25-3.561C13.292.879 14.557.16 15.769 0zM20.67 16.716v.045c-.455 1.378-1.104 2.559-1.896 3.655-.723.995-1.609 2.334-3.191 2.334-1.367 0-2.275-.879-3.676-.903-1.482-.024-2.297.735-3.652.926h-.462c-.995-.144-1.798-.932-2.383-1.642-1.725-2.098-3.058-4.808-3.306-8.276v-1.019c.105-2.482 1.311-4.5 2.914-5.478.846-.52 2.009-.963 3.304-.765.555.086 1.122.276 1.619.464.471.181 1.06.502 1.618.485.378-.011.754-.208 1.135-.347 1.116-.403 2.21-.865 3.652-.648 1.733.262 2.963 1.032 3.723 2.22-1.466.933-2.625 2.339-2.427 4.74.176 2.181 1.444 3.457 3.028 4.209z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      [
        {
          "color": "#FFF",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Entrar com Apple
  </Text>
  <View />
</View>
`;
